File: model/doctorModel.js
const mongoose = require("mongoose")
const Schema = mongoose.Schema;   
const ObjectId = mongoose.Types.ObjectId


const DoctorSchema = new Schema({
    email : { type: String, unique: true, required: true },
    password : { type: String, required: true },
    name : { type: String, required: true },
    phone : { type: String, required: true},
    licenceId : { type: String, required: true, unique: true },
    availTime : { type: String, required: true },
    fees : { type: Number, required: true },
    speciality : {type : String, required : true},
    role : { type : String, default : "doctor"}

},{timestamps:true});



const DoctorModel = mongoose.model('doctors', DoctorSchema)


module.exports = {
    DoctorModel
}

File: medapp_frontend/lib/models/doctor_model.dart
class DoctorModel {
  final String id;
  final String name;
  final String email;
  final String phone;
  final String licenceId;
  final String availTime;
  final int fees;
  final String speciality;
  final String? fcmToken;

  DoctorModel({
    required this.id,
    required this.name,
    required this.email,
    required this.phone,
    required this.licenceId,
    required this.availTime,
    required this.fees,
    required this.speciality,
    this.fcmToken,
  });

  factory DoctorModel.fromJson(Map<String, dynamic> json) {
    return DoctorModel(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'] ?? '',
      licenceId: json['licenceId'] ?? '',
      availTime: json['availTime'] ?? '',
      fees: json['fees'] ?? 0,
      speciality: json['speciality'] ?? '',
      fcmToken: json['fcmToken'],
    );
  }
}
